### PR TITLE
chore: add mcp build pipeline script

### PR DESCRIPTION
**Pull request by blaisedias**
Creat a pipleline that can be parameterized by commit IDs
will (re)build images and install bundles for mcp tagged
by commit ID

Essentially the "build" part of the nightly pipeline

### CODE REVIEW

File: pipelines/Mayastor-QA-mcp-build/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        script {
+          common = load "./pipelines/common/common.groovy"
+          build_params = [
+            mayastorBranch : params.mayastorBranch,
+            mayastorRev : params.mayastorRev,
+            mcpBranch : params.mcpBranch,
+            mcpRev : params.mcpRev,
+            imageTag : params.imageTag,
+          ]
+          if (!build_params['imageTag']?.trim()) {
```
Comment (chriswldenyer) [inline pipelines/Mayastor-QA-mcp-build/Jenkinsfile:41]: Isn't this already trimmed by the parameter definition?

File: pipelines/Mayastor-QA-mcp-build/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    string(defaultValue: '', name: 'mcpRev', trim: true, description: 'mcp revision')
+    string(defaultValue: '', name: 'imageTag', trim: true, description: 'docker image tag')
+  }
+  stages {
+    stage('build docker images') {
+      agent { label 'nixos-mayastor' }
+      steps {
+        script {
+          common = load "./pipelines/common/common.groovy"
+          build_params = [
```
Comment (chriswldenyer) [inline pipelines/Mayastor-QA-mcp-build/Jenkinsfile:34]: although not necessary here, maybe we should get in the habit of using 'def' for all local parameters.

File: pipelines/common/common.groovy [hunk 3]
```diff
@@ -142,6 +142,59 @@ def BuildMCPImages(Map parms) {
 }
 
 
+def BuildMCPRevisionImages(Map params) {
```
Comment (chriswldenyer) [inline pipelines/common/common.groovy:145]: There is a lot of code in common with BuildMCPImages().
I suppose ultimately that function could just call this one with empty revision parameters.

**Review by chriswldenyer — APPROVED** [review]
LGTM, some comments.

File: pipelines/Mayastor-QA-mcp-build/Jenkinsfile (hunk 2 above)
Comment (blaisedias) [reply pipelines/Mayastor-QA-mcp-build/Jenkinsfile:34]: Done.

File: pipelines/common/common.groovy (hunk 3 above)
Comment (blaisedias) [reply pipelines/common/common.groovy]: Replaced code in `BuildMCPImages` with that from  `BuildMCPRevisionImages`  and removed the latter.
Tested with https://mayastor-ci.mayadata.io/job/blaise-test/job/mcp-build-pipeline/

File: pipelines/common/common.groovy [hunk 4]
```diff
@@ -102,36 +102,50 @@ def BuildImages(mayastorBranch, moacBranch, test_tag) {
...
+  }
   sh "cd Mayastor && git submodule update --init"
+  sh "cd Mayastor && git status"
 
   // Build images (REGISTRY is set in jenkin's global configuration).
   // Note: We might want to build and test dev images that have more
   // assertions instead but that complicates e2e tests a bit.
   // Build mayastor and mayastor-csi
-  sh "cd Mayastor && ./scripts/release.sh $build_flags --registry \"${env.REGISTRY}\" --alias-tag \"$test_tag\" "
+  sh "cd Mayastor && ./scripts/release.sh --registry \"${env.REGISTRY}\" --alias-tag \"$test_tag\" "
```
Comment (chriswldenyer) [inline pipelines/common/common.groovy:139]: Do the mayastor and CP builds not need build_flags for coverage?

**Review by chriswldenyer — APPROVED** [review]
LGTM but one question

File: pipelines/common/common.groovy [hunk 5]
```diff
@@ -102,36 +102,50 @@ def BuildImages(mayastorBranch, moacBranch, test_tag) {
...
 
-  // Build mayastor control plane
+  // Build moac
```
Comment (rohan2794) [inline pipelines/common/common.groovy:141]: are we building MOAC as control plane ?

**Review by rohan2794 — APPROVED** [review]
LGTM

File: pipelines/common/common.groovy (hunk 5 above)
Comment (blaisedias) [reply pipelines/common/common.groovy]: fixed comment

File: pipelines/common/common.groovy (hunk 4 above)
Comment (blaisedias) [reply pipelines/common/common.groovy]: Doh! copy and paste. Fixed, we must always use `buildflags`

**Review by chriswldenyer — APPROVED** [review]
LGTM

**Pull request merged**